src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -126,6 +128,23 @@ protected override string GenerateCommandLineCommands ()
 		{
 			var cmd = GetCommandLineBuilder ();
 
+			if (NamespaceTransforms?.Any () == true) {
+				// ex: obj/Debug/generated/msbuild-metadata.xml
+				var transform_file = Path.Combine (OutputDirectory, "..", "msbuild-metadata.xml");
+
+				var xml = new XDocument ();
+				var root = new XElement ("metadata");
+				xml.Add (root);
+
+				foreach (var nt in NamespaceTransforms)
+					xml.Root.Add (new XElement ("ns-replace", new XAttribute ("source", nt.ItemSpec), new XAttribute ("replacement", nt.GetMetadata ("replacement"))));
```

Comment:
Should we emit a warning/error if `%(Replacement)` is blank? I'm just thinking if someone put `Replace="AndroidX"` on accident, they won't know what's wrong?

Reply:
`%(Replacement)` being *blank* is valid and useful.  I guess the question is should we require them to provide the blank attribute `Replacement=""`?  Perhaps we should.

Reply:
Added and documented new error.